### ISSUE TITLE
Complete normalization modules

### ISSUE THREAD

**Issue by tom-huntington** [pytorch/pytorch#76613]
### 🚀 The feature, motivation and pitch

For input of shape `(N, C, L)`, batch norm lets you normalize over `N`, and `L`, layer norm lets you normalize over `C`, and `L`. However, you are missing a module that allows you to normalize over `N` and `L`.

I recently came accross this twice, once when wanting to normalize input with something `nn.functional` and just now when I realized that to do batch norm with transformers since the channels are actually in the `-1`th dimension i.e. `(N, L, C)` you need this missing normalization.

![](https://pic1.zhimg.com/v2-27977bfda164cc78cbef54c5da70d503_1440w.jpg)

### Alternatives

You can just transpose, then do batch norm then transpose back but I doubt the performance will be as good.

**Comment by jbschlosser** [comment]
Hey @tom-huntington, thanks for the request! If you're using PyTorch's [Transformer](https://pytorch.org/docs/stable/generated/torch.nn.Transformer.html), the default expected format is `(L, N, C)` with `batch_first=False`, and you should be able to normalize over N and C by passing a 2D `normalized_shape` to [`nn.LayerNorm`](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html).

Hopefully that helps, but otherwise I would recommend just transposing, as I don't think we want to add yet another normalization layer simply to avoid the transposes.

**Comment by tom-huntington** [comment]
> as I don't think we want to add yet another normalization layer simply to avoid the transposes.

I understand. Just seems a bit weird you don't provide a way to normalize over arbitrary dimensions, and you have to hack something together with transpose. But maybe there's implementation/performance reasons.
